File: src/main/java/rag/orchestrator/RagOrchestrator.java
package main.java.rag.orchestrator;

import main.java.rag.config.Config;
import main.java.rag.context.Context;
import main.java.rag.trace.JsonlTraceSink;
import main.java.rag.trace.TraceBus;

public class RagOrchestrator {
    private final Context context;
    private final TraceBus traceBus;

    public RagOrchestrator(Context context) {
        this.context = context;
        this.traceBus = new TraceBus();
    }
     
    public void run(Config config, String question) {
        // Register JsonlTraceSink for logging
        JsonlTraceSink jsonlSink = new JsonlTraceSink(config.getLogsDirPath());
        traceBus.register(jsonlSink);
        
        
        RagPipeline pipeline = new SequentialRagPipeline(config, context, traceBus);
        pipeline.execute();
        
    }
}

File: iteration1_py/src/orchestrator/rag_orchestrator.py
from pathlib import Path

from src.config.config import Config
from src.context.context import Context
from src.orchestrator.sequential_rag_pipeline import SequentialRagPipeline
from src.trace.trace_bus import TraceBus
from src.trace.jsonl_trace_sink import JsonlTraceSink


class RagOrchestrator:
    
    def __init__(self, context: Context):
       
        self.context = context
        self.trace_bus = TraceBus()
    
    def run(self, config: Config, question: str) -> None:
       
        # Register JsonlTraceSink for logging
        jsonl_sink = JsonlTraceSink(config.get_logs_dir_path())
        self.trace_bus.register(jsonl_sink)
        
        # Create and execute pipeline
        pipeline = SequentialRagPipeline(config, self.context, self.trace_bus)
        pipeline.execute()
